### ISSUE TITLE
Specify cargo build target dir?

### ISSUE THREAD

**Issue by shrinktofit** [rustwasm/wasm-pack#1156]
## 💡 Feature description

I want to specify all the intermediate compilation files or wasm output into specified dir. I used `wasm-pack build <input-dir> --out-dir <wasm-pack-output-dir> -- --target-dir <cargo-target-dir>`, but the following error was emitted:

```txt
[INFO]: Installing wasm-bindgen...
error: failed reading '<input-dir>\target\wasm32-unknown-unknown\release\<not-important>.wasm'

Caused by:
    系统找不到指定的路径。 (os error 3)
    ^^^^^^^^^^^^^^^
    (Which means "system can not find specified path")
```

Seems like wasm-pack always assumes the cargo target output is emitted to `<input-dir>/target`?

#### 💻 Basic example
Include a basic code example if possible. Omit this section if not applicable.

**Comment by kaimast** [comment]
To me it looks like as wasm-pack changes directories incorrectly.

I specify `CARGO_TARGET_DIR=wasm-target` and it ends up compiling everything in the following folder: `<workspace_dir>/<crate_dir>/wasm-target` (which isn't the correct path).

But then wasm-pack looks for the binary in `<workspace_dir>/wasm-target` (which is the correct path) and fails.

**Comment by drager** [comment]
Fixed in #1331.
